ARCHITECTURE.md

```diff
@@ -0,0 +1,154 @@
+![BluepillIcon](doc/img/bluepill_text.png)
+
+![](https://github.com/linkedin/bluepill/workflows/master/badge.svg)
+
+
+## Bluepill Build
+- `bluepill.sh` builds Bluepill in CI (It has a sample app)
+	- Technically the script builds Bluepill & the sample app, and runs the tests
+- ``BPSampleApp` is the test bed we use for testing Bluepill functionalities. It contains test targets that simulate real world scenarios, e.g a test that crashes, timeouts, UI blockers etc.
+	- X86_64 arch is for iOS simulators running on Intel based Macs. arm64 arch is for iOS device and Apple Silicon based Macs.
+	- For development we usually build only 1 architecture for the target device / simulator. For distribution we would build multi-arch archives to cover the variety of devices.
+	- See [Fat Binary](https://en.wikipedia.org/wiki/Fat_binary) for reference.
+- Side note: iOS is a variant of MacOS and Android ships a VM with a Linux based Android kernel
+- Originally Apple Emulators translated ARM -> `x86_64` but that was very slow!
+	- Helped test iOS apps in Mac by giving you an `x86_64` executable
+- Now they compile all frameworks for `x86_64` and give you a sandbox in which you can build a native binary
+	- It's faster to get an exectuable to test this way
+
+
+## Some notes on `Bundles`
+- App (builds with) -> (Xcode/Bazel) -> `X.app` (packaged into an IPA, an archive of the app or "Bundle")
+- iPhones have ARM processors while Macs have `x86_64` (pre M1 at least)
+- Compilation results in an architecture specific binary (or app binary)
+
+## How iOS Simulators work in Mac OS and some tips for verification
+- `xcrun simctl boot` will boot the simulator (Alternatively run Simulator.App in /Applications/Xcode.App)
+- `ps axuw | grep 'launchd_sim'` -> This is the `launchd` process running inside the Simulator
+	- Since the Simulator isn't a VM (like Android) and shares the same Kernel as the underlying Mac OS, it's possible to inspect the processes running inside the simulator
```

Comment:
```suggestion
	- Since the Simulator isn't a VM and shares the same Kernel as the underlying Mac OS, it's possible to inspect the processes running inside the simulator
```